profiler/data.py

```diff
@@ -0,0 +1,137 @@
+import os
+import pickle
+import re
+from datetime import date, datetime
+from typing import Optional
+
+import somacore
+
+import tiledbsoma
+
+
+class ProfileData:
+    """This class represent the dat stored per run"""
+
+    date = str(date.today())
+    now = str(datetime.now())
+    idx = now.find(".")
+    if idx != -1:
+        now = now[: idx - 1]
+    somacore_version = somacore.__version__
+    tiledbsoma_version = tiledbsoma.__version__
+
+    def __init__(
+        self,
+        process: str,
+        p1out: Optional[str],
+        p2out: Optional[str],
+        rt: float,
+        ut: float,
+        st: float,
+        max_set_size: int,
+        page_reclaims: int,
+        page_faults: int,
+        cycles_elapsed: int,
+        peak_memory: int,
+        other_stats: str = None,
+    ):
+        self.process = process
+        self.rt = rt
+        self.ut = ut
+        self.st = st
+        self.max_set_size = max_set_size
+        self.page_reclaims = page_reclaims
+        self.page_faults = page_faults
+        self.peak_memory = peak_memory
+        self.cycles_elapsed = cycles_elapsed
+        self.process = process
+        self.p1out = p1out
+        self.p2out = p2out
+        self.other_stats = other_stats
+
+    def __repr__(self):
+        return (
+            f"{self.__class__} process {self.process} rt {self.rt} ut {self.ut} st {self.st} "
+            f"max mem {self.max_set_size} p1out {self.p1out} p2out {self.p2out} other_stats {self.other_stats} "
+            f"somacore_v {self.somacore_version} tiledbsoma_v {self.tiledbsoma_version}"
+        )
+
+
+def extract_key_from_filename(filename):
+    match = re.match("(.+)\.run", filename)
+    assert match
+    return match.groups()[0]
+
+
+class ProfilerDict:
```

Comment:
Any reason why `ProfilerDict` and `ProfileDB` can't be combined into one class? They seem to both be implementing the file-based db. 

For accommodate future, alternate implementions, I would recommend adding an abstract class that defines the interface for any database implementation we might want to support (file-based, SQLlite, etc.), as `class ProfilerDB(ABC): ...` And then merge these two classes into a single class as `class FileSystemProfilerDB(ProfilerDB): ...`.  See https://docs.python.org/3/library/abc.html and https://realpython.com/lessons/abstract-classes/.

Reply:
I had a similar idea but wanted any new implementation that implement a persistent Dict class (like SQLiteDict) can be used instead of `ProfilerDict`. As you can see ProfileDB does not really care how `ProfilerDict` is implemented. To be more clear I could call ProfileDict `FileSystemProfilerDict`. LMK what you think @atolopko-czi